From b5e1a829fc40d7795024c1f27867a5aee9f13957 Mon Sep 17 00:00:00 2001
From: Heiko Becker <heirecka@exherbo.org>
Date: Sun, 29 Jan 2023 23:01:23 +0100
Subject: [PATCH] Fix build with gcc 13 by including <cstdint>

Like other versions before, gcc 13 moved some includes around and as a
result <cstdint> is no longer transitively included. Explicitly include
it for uint*_t.

Upstream: https://github.com/xbmc/xbmc/pull/23314
Signed-off-by: Bernd Kuhls <bernd.kuhls@t-online.de>
---
 xbmc/cores/RetroPlayer/cheevos/Cheevos.h                     | 1 +
 xbmc/cores/VideoPlayer/VideoRenderers/ColorManager.cpp       | 1 +
 xbmc/cores/VideoPlayer/VideoRenderers/ColorManager.h         | 1 +
 xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/VaapiEGL.h | 1 +
 xbmc/pictures/Picture.h                                      | 2 ++
 xbmc/platform/Filesystem.h                                   | 1 +
 xbmc/platform/posix/Filesystem.cpp                           | 1 +
 xbmc/utils/CSSUtils.cpp                                      | 1 +
 xbmc/utils/CharArrayParser.cpp                               | 1 +
 xbmc/utils/CharArrayParser.h                                 | 1 +
 xbmc/utils/StreamUtils.h                                     | 1 +
 xbmc/windowing/X11/GLContextEGL.h                            | 2 ++
 12 files changed, 14 insertions(+)

diff --git a/xbmc/cores/RetroPlayer/cheevos/Cheevos.h b/xbmc/cores/RetroPlayer/cheevos/Cheevos.h
index 71c5ed5896..70ffe41ed8 100644
--- a/xbmc/cores/RetroPlayer/cheevos/Cheevos.h
+++ b/xbmc/cores/RetroPlayer/cheevos/Cheevos.h
@@ -10,6 +10,7 @@
 
 #include "RConsoleIDs.h"
 
+#include <cstdint>
 #include <map>
 #include <string>
 
diff --git a/xbmc/cores/VideoPlayer/VideoRenderers/ColorManager.cpp b/xbmc/cores/VideoPlayer/VideoRenderers/ColorManager.cpp
index 0c0d953aca..e1888814cf 100644
--- a/xbmc/cores/VideoPlayer/VideoRenderers/ColorManager.cpp
+++ b/xbmc/cores/VideoPlayer/VideoRenderers/ColorManager.cpp
@@ -15,6 +15,7 @@
 #include "utils/TimeUtils.h"
 #include "utils/log.h"
 
+#include <cstdint>
 #include <math.h>
 #include <vector>
 
diff --git a/xbmc/cores/VideoPlayer/VideoRenderers/ColorManager.h b/xbmc/cores/VideoPlayer/VideoRenderers/ColorManager.h
index b23da958d9..1366c3a1d3 100644
--- a/xbmc/cores/VideoPlayer/VideoRenderers/ColorManager.h
+++ b/xbmc/cores/VideoPlayer/VideoRenderers/ColorManager.h
@@ -12,6 +12,7 @@
 #include <lcms2.h>
 #endif
 
+#include <cstdint>
 #include <string>
 
 extern "C"
diff --git a/xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/VaapiEGL.h b/xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/VaapiEGL.h
index 994a4d063b..a8102cc78d 100644
--- a/xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/VaapiEGL.h
+++ b/xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/VaapiEGL.h
@@ -9,6 +9,7 @@
 #pragma once
 
 #include <array>
+#include <cstdint>
 
 #if defined(HAS_GL)
 // always define GL_GLEXT_PROTOTYPES before include gl headers
diff --git a/xbmc/pictures/Picture.h b/xbmc/pictures/Picture.h
index 2d469d795d..ee5a270d86 100644
--- a/xbmc/pictures/Picture.h
+++ b/xbmc/pictures/Picture.h
@@ -11,6 +11,8 @@
 #include "pictures/PictureScalingAlgorithm.h"
 #include "utils/Job.h"
 
+#include <cstddef>
+#include <cstdint>
 #include <string>
 #include <vector>
 
diff --git a/xbmc/platform/Filesystem.h b/xbmc/platform/Filesystem.h
index 3e6b7ec52b..732be9294c 100644
--- a/xbmc/platform/Filesystem.h
+++ b/xbmc/platform/Filesystem.h
@@ -8,6 +8,7 @@
 
 #pragma once
 
+#include <cstdint>
 #include <string>
 #include <system_error>
 namespace KODI
diff --git a/xbmc/platform/posix/Filesystem.cpp b/xbmc/platform/posix/Filesystem.cpp
index 477e7b660f..6e7282ca00 100644
--- a/xbmc/platform/posix/Filesystem.cpp
+++ b/xbmc/platform/posix/Filesystem.cpp
@@ -19,6 +19,7 @@
 #include <sys/statfs.h>
 #endif
 
+#include <cstdint>
 #include <cstdlib>
 #include <limits.h>
 #include <string.h>
diff --git a/xbmc/utils/CSSUtils.cpp b/xbmc/utils/CSSUtils.cpp
index 96105878e0..329d70cdbc 100644
--- a/xbmc/utils/CSSUtils.cpp
+++ b/xbmc/utils/CSSUtils.cpp
@@ -8,6 +8,7 @@
 
 #include "CSSUtils.h"
 
+#include <cstdint>
 #include <string>
 
 namespace
diff --git a/xbmc/utils/CharArrayParser.cpp b/xbmc/utils/CharArrayParser.cpp
index 2e917e2ff2..5aeec2040b 100644
--- a/xbmc/utils/CharArrayParser.cpp
+++ b/xbmc/utils/CharArrayParser.cpp
@@ -10,6 +10,7 @@
 
 #include "utils/log.h"
 
+#include <cstdint>
 #include <cstring>
 
 void CCharArrayParser::Reset()
diff --git a/xbmc/utils/CharArrayParser.h b/xbmc/utils/CharArrayParser.h
index 7e5e314161..3430946dd3 100644
--- a/xbmc/utils/CharArrayParser.h
+++ b/xbmc/utils/CharArrayParser.h
@@ -8,6 +8,7 @@
 
 #pragma once
 
+#include <cstdint>
 #include <string>
 
 /*!
diff --git a/xbmc/utils/StreamUtils.h b/xbmc/utils/StreamUtils.h
index 079cfa3f5c..3d16e8a222 100644
--- a/xbmc/utils/StreamUtils.h
+++ b/xbmc/utils/StreamUtils.h
@@ -8,6 +8,7 @@
 
 #pragma once
 
+#include <cstdint>
 #include <string>
 
 static constexpr int MP4_BOX_HEADER_SIZE = 8;
diff --git a/xbmc/windowing/X11/GLContextEGL.h b/xbmc/windowing/X11/GLContextEGL.h
index afea0b7c43..441787b671 100644
--- a/xbmc/windowing/X11/GLContextEGL.h
+++ b/xbmc/windowing/X11/GLContextEGL.h
@@ -12,6 +12,8 @@
 #include "system_egl.h"
 #include "threads/CriticalSection.h"
 
+#include <cstdint>
+
 #include <EGL/eglext.h>
 #ifdef HAVE_EGLEXTANGLE
 #include <EGL/eglext_angle.h>
-- 
2.39.2

